File: selfstudy/python/cities_min.py
'''
task 173:
Cities - Min
You need to find the minimum value among city populations in thousands
'''

def lowest_cities(populations):
    return min(populations) if populations else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], lowest_cities([10, 20, 30])),
        ([50, 60, 70], lowest_cities([50, 60, 70])),
        ([1, 2, 3], lowest_cities([1, 2, 3])),
        ([100, 200], lowest_cities([100, 200])),
        ([], lowest_cities([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = lowest_cities(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# works i think

File: selfstudy/golang/cities_min.go
/*
task 273:
Cities - Min
You need to find the minimum value among city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestCities(populations []int) int {
	if len(populations) == 0 { return 0 }
	m := populations[0]
	for _, v := range populations {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestCities([]int{1, 2, 3})},
		{[]int{100, 200}, LowestCities([]int{100, 200})},
		{[]int{}, LowestCities([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
